### DISCUSSION TITLE
Conditional job execution for a specific label name

### DISCUSSION THREAD

**Discussion by mollyIV** [community/community#26160] [Actions]
<p>Hello :wave:</p>
<p>I am trying to execute a job only when a specific label is added to a pull request:</p>
<pre><code>name: Notifications
on:
  pull_request:
    types: [labeled]

jobs:
  job:
   if: contains( github.event.pull_request.labels.*.name, ${{ 'status: needs-attention' }})
</code></pre>
<p>Unfortunately I am getting an error:</p>
<pre><code>incomplete explicit mapping pair; a key node is missed; or followed by a non-tabulated empty line
</code></pre>
<p>It seems that a space after the colon character in a label name bothers the workflow. </p>
<p>Could anyone please help me to figure that out? Thanks :bowing_man:‍♂</p>

**Comment by Yanjingzhu** [comment]
<p>What’s the specific label name?  <strong>status: needs-attention</strong>?</p>
<p>Based on my test, it is not colon but <strong>space</strong> fail the workflow. </p>
<p>If contains() function doesn’t allow including a space in the label name. I tested add ${{ }} around the string, and try to add \ before the space.  This will make the contains() function invalid. </p>
<p>If you remove the space from your lable and make it   <strong>status:needs-attention, </strong> you could use </p>
<pre><code>if: contains( github.event.pull_request.labels.*.name, 'status:needs-attention' )
</code></pre>

**Comment by mollyIV** [comment]
<p>Escaping a space after a colon in a label name does a trick :sparkles:</p>
<pre><code>jobs:
  job:
   if: contains( github.event.pull_request.labels.*.name, ${{ 'status:\ needs-attention' }})
</code></pre>
<p>Thanks for your help <a class="mention" href="/yanjingzhu">@yanjingzhu</a> 🙇‍♂️</p>

**Comment by Yanjingzhu** [comment]
<p><a class="mention" href="/mollyiv">@mollyiv</a> Hi, I tested escaping a space before, but the if condition is always true whatever the label name is. Please see my logs, I only have a label with name “123”.</p>
<p><strong>If </strong>** contains(github.event.pull_request.labels.*.name, ${{ ‘status:\ needs-attention’ }} )** returns true. </p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/6/60de0c3ca742250363aca059a86f2e0bd93ef96b.png" data-download-href="/uploads/short-url/dOVzueyVHD0Ccn4ZIn3mCApfJzB.png?dl=1" title="label filter.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/24806478/181095864-1297b3db-8f99-4a5f-ade7-0470676e08a6.png" alt="label filter.png" data-base62-sha1="dOVzueyVHD0Ccn4ZIn3mCApfJzB" width="472" height="500" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/6/60de0c3ca742250363aca059a86f2e0bd93ef96b_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">label filter.png</span><span class="informations">779×825 27.9 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Please check whether you have the same issue. </p>

**Comment by mollyIV** [comment]
<p>Hey <a class="mention" href="/yanjingzhu">@yanjingzhu</a> ,</p>
<p>thanks for your investigation, I confirm that I have the very same issue. Didn’t really expect that. Wondering what you can do about it :thinking: I’ve been trying plenty escaping from <a href="https://github.com/ansible/ansible/issues/2769" rel="nofollow noopener">https://github.com/ansible/ansible/issues/2769</a> but no luck. </p>

**Comment by Yanjingzhu** [comment]
<p>I am trying to get help from engineering team , asking them the right way to filter a label name with space. I will update here at the first time when I get any good news. </p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p><a class="mention" href="/mollyiv">@mollyiv</a>  You could add “” around the whole contains() function to make it work without escaping space. yaml is a superset of json so wrapping with double quotes should help (since the condition contains single quotes). </p>
<pre><code>if: "contains(github.event.pull_request.labels.*.name, 'status: needs-attention')"
</code></pre>
<p>or this should work too</p>
<pre><code>if: |
  contains(github.event.pull_request.labels.*.name, 'status: needs-attention')
</code></pre>
<p>  </p>

**Comment by mollyIV** [comment]
<p>I just finished testing your suggestions and both do the job perfectly - the job is triggered <strong>only</strong>  when a new label was added and a pull request contains <em>status: needs-attention</em> as one of its labels :raised_hands:</p>
<p>Marking your response as a solution :white_check_mark:</p>
<p>Thank you sooo much for your help <a class="mention" href="/yanjingzhu">@yanjingzhu</a> 🙇‍♂️</p>
